### PR TITLE
fix(likes): ajax load liking users and show in lightbox

### LINKED ISSUE #4797: likes: Prefer loading popup content via ajax (Trac #4797)
_Original ticket http://trac.elgg.org/ticket/4797 on 42605696-10-11 by ewinslow, assigned to unknown._

Elgg version: 1.8

Right now likes has a popup function but the content is all present in the DOM before being shown. I propose we switch these to be real links to urls that show the list of people who liked the item, but then add onclick handlers to prevent navigation and fetch the content via ajax instead. This could really help with initial page load performance.

Note this is slightly different than #3809, which suggested using a real url *and* including the content in the initial dom load. I'm suggesting we use a real URL and omit the content from the initial dom load, especially if generating than content requires a DB fetch (which it does, in the case of likes).

- [x] load content in a lightbox
- [x] load content via ajax

### PR DESCRIPTION
**Pull request by jdalsem**
fixes #4797

### CODE REVIEW

File: mod/likes/views/default/likes/css.php [hunk 1]
```diff
@@ -7,9 +7,8 @@
 /* ***************************************
 	Likes
 *************************************** */
-.elgg-likes {
+.elgg-likes-popup {
```
Comment (ewinslow) [inline mod/likes/views/default/likes/css.php]: Why was this changed? Doesn't this effectively change API if a theme was using the previous class?

File: mod/likes/views/default/likes/popup.php [hunk 2]
```diff
@@ -0,0 +1,17 @@
+<?php
+
+$guid = get_input("guid");
+
+elgg_entity_gatekeeper($guid);
```
Comment (ewinslow) [inline mod/likes/views/default/likes/popup.php]: What happens if likes is requested and this is triggered? What shows up in the lightbox?

**Comment by ewinslow** [comment]
Thanks for doing this, seems like it will be really awesome for perf too.

Can you post screenshots?

**Comment by ewinslow** [comment]
I bet you could even call this out as a `perf` change.

File: mod/likes/views/default/likes/count.php [hunk 3]
```diff
@@ -19,19 +23,11 @@
 	$params = array(
 		'text' => $likes_string,
 		'title' => elgg_echo('likes:see'),
-		'rel' => 'popup',
-		'href' => "#likes-$guid",
-		'class' => 'elgg-non-link',
+		'class' => 'elgg-lightbox',
+		'href' => "ajax/view/likes/popup?guid=$guid",
```
Comment (mrclay) [inline mod/likes/views/default/likes/count.php]: I'd prefer to have a data-view attribute. If JS breaks for some reason (or user clicks this early), the user will be sent to an ugly partial page.

Comment (mrclay) [inline mod/likes/views/default/likes/count.php]: Or just move this this to the colorbox-opts object.

File: mod/likes/views/default/likes/popup.php (hunk 2 above)
Comment (mrclay) [inline mod/likes/views/default/likes/popup.php]: Seems like we just want something like
```php
if (!get_entity($guid)) {
    return;
}
```

File: mod/likes/views/default/likes/css.php (hunk 1 above)
Comment (jdalsem) [inline mod/likes/views/default/likes/css.php]: elgg-likes is no longer used and replaced by popup. If we would reuse elgg-likes we would have conflicting css (as it declared position absolute; which causes issues in the lightbox)

File: mod/likes/views/default/likes/popup.php (hunk 2 above)
Comment (jdalsem) [inline mod/likes/views/default/likes/popup.php]: that means invalid input is not handled, at least we should provide a system_message in that case

File: mod/likes/views/default/likes/count.php (hunk 3 above)
Comment (jdalsem) [inline mod/likes/views/default/likes/count.php]: i think colorbox opts is a good idea

File: mod/likes/views/default/likes/popup.php [hunk 4]
```diff
@@ -0,0 +1,17 @@
...
+
+$list = '<div class="elgg-likes-popup">';
+$list .= elgg_list_annotations(array(
+	'guid' => $guid,
+	'annotation_name' => 'likes',
+	'limit' => 99,
+	'preload_owners' => true,
+	'pagination' => false,
+));
+$list .= '</div>';
```
Comment (juho-jaakkola) [inline mod/likes/views/default/likes/popup.php]: I'd prefer having starting and ending tag on the same row for readability.
```php
echo "<div class=\"elgg-likes-popup\">$list</div>";
```

Comment (hypeJunction) [inline mod/likes/views/default/likes/popup.php]: elgg_format_element() is far more readable.

File: mod/likes/views/default/likes/css.php (hunk 1 above)
Comment (ewinslow) [inline mod/likes/views/default/likes/css.php]: Gotcha

**Comment by jdalsem** [comment]
addressed the comments and changed the commit to `perf`

File: mod/likes/views/default/likes/popup.php [hunk 5]
```diff
@@ -0,0 +1,17 @@
+<?php
+
+$guid = get_input("guid");
+
+if (!get_entity($guid)) {
+	die(elgg_echo("error:missing_data"));
```
Comment (mrclay) [inline mod/likes/views/default/likes/popup.php]: Seems better to echo that and `return;`

**Comment by jdalsem** [comment]
returned instead of died + screenshots for @ewinslow 
screenshot likes popup

**Comment by ewinslow** [comment]
Beautiful

**Pull request merged**